src/adapter/scriptSkipper.ts

```diff
@@ -65,37 +76,156 @@ export class ScriptSkipper {
+    this._sendBlackboxPatterns(blackboxPatterns);
+  }
+
+  private _updateAllScriptsToBeSkipped(): void {
+    const urlsToSkip: string[] = [];
+    for (const entry of this._isUrlSkippedMap.entries()) {
+      if (entry[1]) {
+        urlsToSkip.push(entry[0]);
+      }
+    }
+    this._updateBlackboxedUrls(urlsToSkip);
+  }
+
+  private _sendBlackboxPatterns(patterns: string[]) {
+    for (const blackboxSender of this._blackboxSenders) {
+      blackboxSender.sendPatterns(patterns);
     }
   }
 
-  public setBlackboxSender(debuggerAPI: cdp.DebuggerApi) {
-    this.blackboxSender = debuggerAPI.setBlackboxPatterns.bind(debuggerAPI);
+  public registerNewBlackBoxSender(target: ITarget, debuggerAPI: cdp.DebuggerApi) {
+    const sender = new BlackBoxSender(debuggerAPI);
+    if (!this._blackboxSenders.has(sender)) {
+      this._blackboxSenders.add(sender);
+    }
   }
 
-  public updateSkippingForScript(localpath: string, url: string): void {
-    if (!this._isUrlSkippedMap.has(url)) {
-      if (localpath) {
-        this._isUrlSkippedMap.set(url, this._testRegex(this._nonNodeInternalRegex, localpath));
+  private _updateSourceContainers(affectedUrls: string[], updatedSkipValue: boolean) {
+    for (const container of this._allSourceContainers) {
+      for (const url of affectedUrls) {
+        const source = container.sourceByUrl(url);
+        if (source) {
+          source._blackboxed = updatedSkipValue;
+        }
+      }
+    }
+  }
+
+  private _setUrlToSkip(url: string, skipValue: boolean): void {
+    this._isUrlSkippedMap.set(pathUtils.forceForwardSlashes(url), skipValue);
+  }
+
+  public setSkippingValueForScripts(urls: string[], skipValue: boolean): void {
+    urls.forEach(url => {
+      this._setUrlToSkip(url, skipValue);
+    });
+
+    this._updateAllScriptsToBeSkipped();
+  }
+
+  public isScriptSkipped(url: string): boolean {
+    return this._isUrlSkippedMap.get(pathUtils.forceForwardSlashes(url)) === true;
+  }
+
+  public initializeNodeInternals(nodeInternalsNames: string[]): void {
+    this._allNodeInternals = nodeInternalsNames.map(name => name + '.js');
+  }
+
+  private _isNodeInternal(url: string): boolean {
+    return (
+      (this._allNodeInternals && this._allNodeInternals.includes(url)) ||
+      this._testRegex('^internal/.+\\.js$', url)
+    );
+  }
+
+  public async updateSkippingValueForScript(source: Source) {
+    const url = source.url();
+    if (!this._isUrlSkippedMap.has(pathUtils.forceForwardSlashes(url))) {
+      // applying sourcemappathoverrides results in incorrect absolute paths
+      // for some sources that don't map to disk (e.g. node_internals), so here
+      // we're checking for whether a file actually corresponds to a file on disk
+      const pathOnDisk = await source.existingAbsolutePath();
+      if (pathOnDisk) {
+        // file maps to file on disk
+        this._setUrlToSkip(url, this._testRegex(this._nonNodeInternalRegex, pathOnDisk));
```

Comment:
Probably faster to store these as actual RegExps instead of dynamically constructing a RegExp each time this is run

Reply:
Done!